client/hotrod-client/src/main/java/org/infinispan/client/hotrod/impl/iteration/SegmentKeyTracker.java

```diff
@@ -40,20 +42,38 @@ public SegmentKeyTracker(DataFormat dataFormat, SegmentConsistentHash segmentCon
       segmentStream.forEach(i -> keysPerSegment.set(i, new HashSet<>()));
    }
 
+   private Set<WrappedByteArray> drainKeys() {
+      Set<WrappedByteArray> trackedKeys = new HashSet<>();
```

Comment:
I would add the keys directly to `ReplKeyTracker`. You are creating a new `HashSet` here and when you instantiate the `ReplKeyTracker`.

Reply:
Since you mentioned...I think I can remove this repl tracker altogether

Reply:
... and you can use `keyOnlyTracker` directly :) instead of creating a new `HashSet`.

Reply:
sure, forgot to remove it

Reply:
changed